### ISSUE TITLE
Admin returns internal server error (occasionnal)

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#1818]
Running latest devel package (packetfence-6.3.9-0.20161111.el7.noarch)

The following error appears in the httpd.admin.error log:
```
[Fri Nov 11 10:58:15.049144 2016] [perl:error] [pid 30302] [client 172.20.20.63:46386] Can't call method "ReadConfig" on unblessed reference at /usr/local/pf/lib/pf/config/cached.pm line 862.\n, referer: https://172.20.155.47:1443/admin/configuration
```

Refreshing the same page returns a 200 and displays properly

**Comment by jrouzierinverse** [comment]
We should just get rid of pf::config::cached since it is not needed any more.
It should take ~1 to get rid of it.

**Comment by julsemaan** [comment]
Had another occurence:

```
[Fri Nov 11 11:23:19.685299 2016] [perl:error] [pid 30302] [client 172.20.20.63:46561] Can't call method "ReadConfig" on unblessed reference at /usr/local/pf/lib/pf/config/cached.pm line 862.\n, referer: https://172.20.155.47:1443/admin/configuration
```

**Comment by julsemaan** [comment]
Also whats the unit on the ~1 

Is it 1 hour or 1 year :P ?

**Comment by jrouzierinverse** [comment]
hour

**Comment by julsemaan** [comment]
And the admin will reload the configstore inifile everytime right, so there won't be any cache for the Config::IniFiles objects ?

**Comment by julsemaan** [comment]
BTW, I had another occurence, and it seems to be happening once every 10-15 requests in the admin for me currently

**Comment by jrouzierinverse** [comment]
Yes we got rid of the issue that was causing the config parsing to have  0(n^2) when a config has lot s of sections. So it should not be so bad.

**Comment by julsemaan** [comment]
That will certainly be out of scope for 6.4 I believe

Any band-aid we can apply for that issue, because it is recurring in my tests and affecting about 1 in 10-15 requests

**Comment by jrouzierinverse** [comment]
It will be faster for me remove pf::config::cached then to try and find a bandaid

**Comment by julsemaan** [comment]
The thing is this is a fairly big change this close to the release.

That can be @extrafu's call.

**Comment by jrouzierinverse** [comment]
If you review commit by commit it is pretty small.

**Comment by julsemaan** [comment]
I see the code changes aren't big, but the possible consequences are, we're in a feature freeze and we aim to release in less than 2 business days and testing is almost completed. I'm sure not sure about the timing of that additional change.

As I said, this can be @extrafu's call and other engineers input is welcome (especially from THE PacketFence engineer)

**Comment by extrafu** [comment]
Let's discuss this in one hour.

**Comment by jrouzierinverse** [comment]
Try the following two commits to see if they fix the issue.
c1d888bb85f8d9dfeda5631e4489d01491470c6c
e8ad68ee4fdae1931ab1907afcea05bee3939b27

**Comment by extrafu** [comment]
Any update on this @julsemaan ? We must get the fix/bandaid in v6.4

**Comment by julsemaan** [comment]
Will try it out and report

**Comment by julsemaan** [comment]
Nothing to report after 10-15 minutes of usage.

I'll close and re-open if necessary

**Comment by julsemaan** [comment]
I'm getting a lot of that error in the webservices when I sync the cluster on 6.4

That makes bin/cluster/sync on 6.4 almost unusable

**Comment by dwlfrth** [comment]
Monitoring it

**Comment by extrafu** [comment]
@dwlfrth @jrouzierinverse needs log info, provide him.

**Comment by jrouzierinverse** [comment]
Fix has been fixed in cb1fb0f35a4e1c8f2697a005074d1751b8122d0f
Let me know is this helps

**Comment by julsemaan** [comment]
pf::config::cached is long gone so this is no longer an issue
